File: data/sample.cpp
/* header block */
// Single-line header comment

int main() {
    int x = 1; // initialize x
    /* block comment in cpp */
    int y = x/**/y; // ensure tokens don't merge
    const char* s = "/* not a comment */"; // string must remain
    return y; /* trailing */
}

File: data/sample.m
/* objc header */
// Obj-C single-line

int main() {
    int a = 1; // initialize a
    /* objc block comment */
    int b = a/**/b; // ensure tokens don't merge
    const char* s = "/* not a comment */"; // string must remain
    return b; /* trailing */
}

